rest-api/data_gen/fake_participant_generator.py

```diff
@@ -574,12 +602,69 @@ def _choose_name(self, answer_map):
+        return [_code_answer(code) for code in random.sample(codes, answer_count)]
+    if int(answer_spec['decimal_answer_count']) > 0:
+      return [{'valueDecimal': round(random.uniform(float(answer_spec['min_decimal_answer']),
+                                                    float(answer_spec['max_decimal_answer'])), 1)}
+              for _ in xrange(answer_count)]
+    if int(answer_spec['integer_answer_count']) > 0:
+      return [{'valueInteger': random.randint(int(answer_spec['min_integer_answer']),
+                                              int(answer_spec['max_integer_answer']))}
+              for _ in xrange(answer_count)]
+    if int(answer_spec['date_answer_count']) > 0:
+      min_date = parse(answer_spec['min_date_answer'])
+      max_date = parse(answer_spec['max_date_answer'])
+      days_diff = (max_date - min_date).days
+      return [{'valueDate': (min_date +
+                             datetime.timedelta(days=random.randint(0, days_diff))).isoformat()}
+              for _ in xrange(answer_count)]
+    if int(answer_spec['datetime_answer_count']) > 0:
+      min_date = parse(answer_spec['min_datetime_answer'])
+      max_date = parse(answer_spec['max_datetime_answer'])
+      seconds_diff = (max_date - min_date).total_seconds()
+      return [{'valueDateTime': (min_date +
+                                 datetime.timedelta(seconds=random.randint(0, seconds_diff)))
+                                 .isoformat()}
+              for _ in xrange(answer_count)]
+    if int(answer_spec['boolean_answer_count']) > 0:
+      return [{'valueBoolean': random.random() < 0.5}]
+    if int(answer_spec['string_answer_count']) > 0:
+      return [{'valueString': ''.join([random.choice(string.lowercase) for _ in xrange(20)])}
+              for _ in xrange(answer_count)]
+    if int(answer_spec['uri_answer_count']) > 0:
+      return [{'valueUri': 'gs://some-bucket-name/%s' %
+               ''.join([random.choice(string.lowercase) for _ in xrange(20)])}
+               for _ in xrange(answer_count)]
+    logging.warn('No answer type found for %s, skipping...' % answer_spec['question_code'])
+    return None
+
+  def _choose_answers_for_other_questions(self, answer_map):
+    for question_code, answer_spec in self._answer_specs.iteritems():
+      num_participants = int(answer_spec['num_participants'])
+      # Skip answering this question based on the percentage of participants that answered it.
+      if random.random() > float(num_participants) / float(self._answer_specs_max_participants):
+        continue
+      num_questionnaire_responses = int(answer_spec['num_questionnaire_responses'])
+      num_answers = int(answer_spec['num_answers'])
+      answer_count = 1
+      if num_answers > num_questionnaire_responses:
+        rand_val = random.random() * num_answers / num_questionnaire_responses
+        if rand_val > 1:
+          # Set the answer count to 2 or more
+          answer_count = int(1 + min(1, rand_val))
```

Comment:
The min function here always returns 1, no?

Reply:
Oops. Changed to max.